### PR TITLE
support time extract

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

File: expression/visitor.go [hunk 1]
```diff
@@ -425,3 +428,10 @@ func (bv *BaseVisitor) VisitWhenClause(w *WhenClause) (Expression, error) {
 	}
 	return w, nil
 }
+
+// VisitExtract implements Visitor
```
Comment (qiuyesuifeng) [inline expression/visitor.go]: Visitor interface.

File: parser/parser.y [hunk 2]
```diff
@@ -1635,7 +1650,7 @@ UnReservedKeyword:
 |	"START" | "GLOBAL" | "TABLES"| "TEXT" | "TIME" | "TIMESTAMP" | "TRANSACTION" | "TRUNCATE" | "UNKNOWN" 
 |	"VALUE" | "WARNINGS" | "YEAR" |	"MODE" | "WEEK" | "ANY" | "SOME" | "USER" | "IDENTIFIED" | "COLLATION"
 |	"COMMENT" | "AVG_ROW_LENGTH" | "CONNECTION" | "CHECKSUM" | "COMPRESSION" | "KEY_BLOCK_SIZE" | "MAX_ROWS" | "MIN_ROWS"
-|	"NATIONAL" | "ROW"
+|	"NATIONAL" | "ROW" | "QUARTER"
 
 NotKeywordToken:
```
Comment (shenli) [inline parser/parser.y]: Add new tokens such as SECOND_MICROSECOND here. Or those tokens can not be used as identifier.

Comment (siddontang) [inline parser/parser.y]: SECOND_MICROSECOND and those are reserved key word and can not be used as identifier.

Comment (shenli) [inline parser/parser.y]: Ok

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**